### ISSUE TITLE
Support controllers over the network

### ISSUE THREAD

**Issue by gvieralopez** [Poohl/joycontrol#10]
I was wondering if it is possible to map the inputs of a device in a different system and send them over the network to the the computer near the Switch with the Bluetooth controller.  [This project](https://github.com/Brikwerk/nxbt) does it, but I couldn't use it because their pairing process is not working in SwitchOS 12.1.0.

**Comment by cazwacki** [comment]
This is possible, I've done it on [a personal project](https://github.com/cazwacki/Switch-Streaming) that uses this. Mine is a lot more rudimentary than that one. Basically, you need to write some code that can read inputs from the controller and turn it into a code that is small so you can send them over the network. I use WebSocket to establish connection between client and computer near the Switch. The host computer receives the code and translates it into a corresponding switch input, then uses Poohl's joycontrol to actually send the input to the Switch.

**Comment by gvieralopez** [comment]
> 
> 
> This is possible, I've done it on [a personal project](https://github.com/cazwacki/Switch-Streaming) that uses this. Mine is a lot more rudimentary than that one. Basically, you need to write some code that can read inputs from the controller and turn it into a code that is small so you can send them over the network. I use WebSocket to establish connection between client and computer near the Switch. The host computer receives the code and translates it into a corresponding switch input, then uses Poohl's joycontrol to actually send the input to the Switch.

Thanks a lot for pointing your project. I will test it this weekend. Any chance you know if it works on SwitchOS 12.1.0?

**Comment by cazwacki** [comment]
Yes, it seems to be working. A friend has been playing Animal Crossing frequently on my Switch even recently, with the software updated. Note that the existing issues in joycontrol (e.g. gyroscope erratic behavior, etc.) will be reflected in my project as well.
